CMakeLists.txt

```diff
@@ -0,0 +1,142 @@
+# Check if we are building a pico-sdk based project
+# (or more exactly: if we just got included in a pico-sdk based project)
+if (PICO_SDK_PATH)
+    # If so, load the relevant CMakeLists-file but don't do anything else
+    include(${CMAKE_CURRENT_LIST_DIR}/utility/rp2/CMakeLists.txt)
+    return()
+endif()
+
+cmake_minimum_required(VERSION 3.15)
+
+# Set the project name to your project name
+project(RF24 C CXX)
+include(cmake/StandardProjectSettings.cmake)
+include(cmake/PreventInSourceBuilds.cmake)
+
+# Link this 'library' to set the c++ standard / compile-time options requested
+add_library(project_options INTERFACE)
+target_compile_features(project_options INTERFACE cxx_std_17)
+add_compile_options(-Ofast -Wall -pthread)
+
+if(CMAKE_CXX_COMPILER_ID MATCHES ".*Clang")
+    option(ENABLE_BUILD_WITH_TIME_TRACE "Enable -ftime-trace to generate time tracing .json files on clang" OFF)
+    if(ENABLE_BUILD_WITH_TIME_TRACE)
+        add_compile_definitions(project_options INTERFACE -ftime-trace)
+    endif()
+endif()
+
+# Link this 'library' to use the warnings specified in CompilerWarnings.cmake
+add_library(project_warnings INTERFACE)
+
+# enable cache system
+include(cmake/Cache.cmake)
+
+# standard compiler warnings
+include(cmake/CompilerWarnings.cmake)
+set_project_warnings(project_warnings)
+
+# sanitizer options if supported by compiler
+include(cmake/Sanitizers.cmake)
+enable_sanitizers(project_options)
+
+# allow for static analysis options
+include(cmake/StaticAnalyzers.cmake)
+
+option(BUILD_SHARED_LIBS "Enable compilation of shared libraries" OFF)
+option(ENABLE_TESTING "Enable Test Builds" OFF) # for end-user projects
+option(ENABLE_FUZZING "Enable Fuzzing Builds" OFF) # for end-user projects
+
+if(ENABLE_TESTING)
+    enable_testing()
+    message("Building Tests.")
+    add_subdirectory(test) # directory doesn't exist, so this does nothing.
+endif()
+
+if(ENABLE_FUZZING)
+    message("Building Fuzz Tests, using fuzzing sanitizer https://www.llvm.org/docs/LibFuzzer.html")
+    add_subdirectory(fuzz_test) # directory doesn't exist, so this does nothing.
+endif()
+
+#####################################
+### Now we actually build the library
+#####################################
+
+# pull in function to copy the appropriate includes.h file from a utility sub directory
+include(cmake/copy_includes.cmake)
+
+# get library info from Arduino IDE's required library.properties file
+include(cmake/GetLibInfo.cmake)
+
+# detect the CPU make and type
+include(cmake/detectCPU.cmake) # sets the variable SOC accordingly
+
+# auto-detect what driver to use
+# auto-detect can be overriden using `cmake .. -D RF24_DRIVER=<supported driver>`
+include(cmake/AutoConfig_RF24_DRIVER.cmake)
+
+#[[ adding the utility sub-directory will
+    1. set variables RF24_DRIVER, SOC, RF24_LINKED_DRIVER, and RF24_DRIVER_SOURCES
```

Comment:
comment is outdated. the SOC var is set by the above cmd `include(cmake/detectCPU.cmake)`

Reply:
fixed